### PR TITLE
fix 3169: Polygon.points property

### PR DESCRIPTION
**Pull request by WarriorOfWire**
The getter for vectorio.Polygon#points was not updated with the data type change of the stored points list.

This moves the implementation to shared_module and updates the data type to reflect the actual state.

### CODE REVIEW

**Comment by FoamyGuy** [comment]
I tried building the system for PyPortal with this branch. I got a successful build and flashed the uf2 to the PyPortal but when I connect to the serial console it freezes after a few seconds. Tried both Mu Serial Console and Putty both on Win7. 

I'm going to try building for a different device.

**Comment by WarriorOfWire** [comment]
Thanks for the additional testing @FoamyGuy!

File: shared-module/vectorio/Polygon.c [hunk 1]
```diff
@@ -56,22 +58,35 @@ static void _clobber_points_list(vectorio_polygon_t *self, mp_obj_t points_tuple
...
     self->on_dirty.obj = NULL;
     _clobber_points_list( self, points_list );
+    VECTORIO_POLYGON_DEBUG("\n");
 }
 
 
 mp_obj_t common_hal_vectorio_polygon_get_points(vectorio_polygon_t *self) {
-    return self->points_list;
+    VECTORIO_POLYGON_DEBUG("%p common_hal_vectorio_polygon_get_points {len: %d, points_list: %p}\n", self, self->len, self->points_list);
+    mp_obj_t list = mp_obj_new_list(0, NULL);
```
Comment (jepler) [inline shared-module/vectorio/Polygon.c:72]: I believe that if you call `mp_obj_new_list(self->len/2, NULL);` then the list will be initially allocated with the correct size, avoiding excess allocations.

Comment (WarriorOfWire) [reply shared-module/vectorio/Polygon.c]: Sure, that's better.

**Pull request merged**